pkg/controller/controller_test.go

```diff
@@ -511,6 +620,42 @@ func provisionMockServerSetupExpectations(identityServer *driver.MockIdentitySer
 	}, nil).Times(1)
 }
 
+func provisionFromSnapshotMockServerSetupExpectations(identityServer *driver.MockIdentityServer, controllerServer *driver.MockControllerServer) {
```

Comment:
Its not incredibly clear what expectations we have here, could you add a comment explaining what expectations these are and what we use them for?

Reply:
Added comments